### PR TITLE
feature - home ui

### PR DESCRIPTION
**Pull request by at-datphung**
done Home UI

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
+        super.setSelected(selected, animated: animated)
+
+        // Configure the view for the selected state
+    }
+
```
Comment (at-daiho) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: WARNING - Lines should not have trailing whitespace.

**Comment by at-daiho** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Comment by at-hieutran** [comment]
@at-datphung

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/Resources/Assets.xcassets/iconHome/back.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "back.png",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
```
Comment (at-haiphung) [inline MyApp/Resources/Assets.xcassets/iconHome/back.imageset/Contents.json]: missed 2x file

File: MyApp/Resources/Assets.xcassets/iconHome/next-2.imageset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "images" : [
+    {
+      "idiom" : "universal",
+      "filename" : "next.png",
+      "scale" : "1x"
+    },
+    {
+      "idiom" : "universal",
+      "scale" : "2x"
```
Comment (at-haiphung) [inline MyApp/Resources/Assets.xcassets/iconHome/next-2.imageset/Contents.json]: missed 2x file

File: MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+//  HomeCollectionViewCell.swift
+//  MyApp
+//
+//  Created by AST on 7/21/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+
+class HomeCollectionViewCell: UICollectionViewCell {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift]: final class

File: MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    @IBOutlet weak var imageView: UIImageView!
+
+    override func awakeFromNib() {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift]: remove unused code

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 6]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        button.setImage(#imageLiteral(resourceName: "search"), for: UIControlState.normal)
+        button.addTarget(self, action: #selector(HomeViewController.searchBarItem), for: UIControlEvents.touchUpInside)
+        button.frame = CGRect(x: 0, y: 0, width: 30, height: 30)
+        let rightNaviBar = UIBarButtonItem(customView: button)
+        self.navigationItem.rightBarButtonItem = rightNaviBar
+
+        let leftNaviBar = UIBarButtonItem(title: App.String.categories, style: .plain, target: self, action: #selector(HomeViewController.categories))
+        self.navigationItem.leftBarButtonItem = leftNaviBar
+    }
+    func initTableView() {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: private, configTableView

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 7]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        tableView.register(UINib(nibName: "HomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTableView")
+
+        tableView.register(UINib(nibName: "TrenddingHomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTrendding")
+
+        tableView.delegate = self
+        tableView.dataSource = self
+    }
+    func initCollectionView() {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: private, configCollectionView

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 8]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        collectionView.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "CellCollection")
+
+        collectionView.dataSource = self
+        collectionView.delegate = self
+    }
+    func searchBarItem() {
+
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: TODO

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 9]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+    }
+    func categories() {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: TODO

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 10]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "CellTableView", for: indexPath) as? HomeTableViewCell else {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: use SwiftUlties

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 11]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        }
+        cell.delegate = self
+        return cell
+    }
+}
+// MARK: - UITableViewDelegate
+extension HomeViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let details = DetailViewController()
+        self.navigationController?.pushViewController(details, animated: true)
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: remove self

File: MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+import UIKit
+
+class ListCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var itemImageView: UIImageView!
+    @IBOutlet weak var itemLabel: UILabel!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        layer.borderWidth = 1
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift]: use SwiftUlties

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+//  HomeTableViewCell.swift
+//  MyApp
+//
+//  Created by AST on 7/21/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+protocol HomeTableViewCellDelegate: class {
+    func HomeTableViewCellDelegate(_ collectionview: UICollectionView, didSelectItemAt indexPath: IndexPath)
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: follow delegate pattern

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+class HomeTableViewCell: UITableViewCell {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: final class

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+}
+extension HomeTableViewCell: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 2
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as? ListCollectionViewCell else {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: use swiftUlties

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 16]
```diff
@@ -10,7 +10,113 @@ import UIKit
 
 class HomeViewController: UIViewController {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: final class

File: MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift [hunk 17]
```diff
@@ -0,0 +1,22 @@
...
+//  ListCollectionViewCell.swift
+//  MyApp
+//
+//  Created by AST on 7/24/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+
+class ListCollectionViewCell: UICollectionViewCell {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift]: final

**Comment by at-hieutran** [comment]
@at-datphung 
Pls fix all comment

**Comment by at-daiho** [comment]
Total linter violations in pull request: 1
Serious: 0
Warnings: 1

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

**Comment by at-hieutran** [comment]
@at-haiphung review giup em voi a.

File: MyApp/View/Base/ViewController.swift [hunk 18]
```diff
@@ -9,4 +9,16 @@
...
 
-class ViewController: UIViewController, MVVM.View { }
+class ViewController: UIViewController, MVVM.View {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        confignavigationBar()
+    }
+
+    private func confignavigationBar() {
+        let navi = self.navigationController?.navigationBar
```
Comment (at-haiphung) [inline MyApp/View/Base/ViewController.swift]: remove self
guard let navigationBar = navigationController?.navigationBar ...

File: MyApp/View/Base/ViewController.swift [hunk 19]
```diff
@@ -9,4 +9,16 @@
...
+        navi?.titleTextAttributes = [NSForegroundColorAttributeName: UIColor.white]
+        navi?.barTintColor = UIColor.black.withAlphaComponent(0.8)
+        navi?.tintColor = UIColor.white
```
Comment (at-haiphung) [inline MyApp/View/Base/ViewController.swift]: No need UIColor.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 20]
```diff
@@ -8,115 +8,149 @@
...
-class HomeViewController: ViewController {
+final class HomeViewController: ViewController {
 
-    @IBOutlet weak var tableView: UITableView!
-    @IBOutlet weak var collectionView: UICollectionView!
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var listCollectionView: UICollectionView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    fileprivate var images: [UIImage] = [#imageLiteral(resourceName: "miquang"), #imageLiteral(resourceName: "monan1"), #imageLiteral(resourceName: "monan2"), #imageLiteral(resourceName: "monan3"), #imageLiteral(resourceName: "monan4"), #imageLiteral(resourceName: "monan5")]
+    private var frame: CGRect = CGRect(x: 0, y: 0, width: 0, height: 0)
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: .zero

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 21]
```diff
@@ -8,115 +8,149 @@
...
         let rightNaviBar = UIBarButtonItem(customView: button)
-        self.navigationItem.rightBarButtonItem = rightNaviBar
+        navigationItem.rightBarButtonItem = rightNaviBar
+    }
+
+    private func configScroolView() {
+        scrollView.delegate = self
+        automaticallyAdjustsScrollViewInsets = false
+        scrollView.contentSize = CGSize(width: headerView.bounds.size.width * CGFloat(images.count), height: headerView.bounds.height)
+        for index in 0..<6 {
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: foreach (index, image) in images.enumerate()

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 22]
```diff
@@ -8,115 +8,149 @@
...
+        listCollectionView.dataSource = self
+    }
+
+    @objc private func imageTapped() {
+        let vc = DetailViewController()
+        navigationController?.pushViewController(vc, animated: true)
+    }
+
+    @objc private func moveToNextPage () {
+        let pageWidth: CGFloat = scrollView.frame.width
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: no need : CGFloat

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 23]
```diff
@@ -8,115 +8,149 @@
...
+            slideToX = 0
+        }
+        scrollView.scrollRectToVisible(CGRect(x: slideToX, y: 0, width: pageWidth, height: self.scrollView.frame.height), animated: true)
     }
-    func initTableView() {
-        tableView.register(UINib(nibName: "HomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTableView")
 
-        tableView.register(UINib(nibName: "TrenddingHomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTrendding")
+    private func moveToBackPage () {
+        let pageWidth: CGFloat = scrollView.frame.width
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: no need CGFloat

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 24]
```diff
@@ -8,115 +8,149 @@
...
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        resetTimer()
     }
 }
 // MARK: - UICollectionViewDataSource
 extension HomeViewController: UICollectionViewDataSource {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-
-        return 3
+        return 4
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: numberOfItems is constant?

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 25]
```diff
@@ -8,115 +8,149 @@
...
-    }
-}
-// MARK: - UITableViewDataSource
-extension HomeViewController: UITableViewDataSource {
-    func numberOfSections(in tableView: UITableView) -> Int {
-        return 2
-    }
+        guard let layout = collectionViewLayout as? UICollectionViewFlowLayout else { fatalError("error") }
+        layout.minimumLineSpacing = 10
+        let numberOfItemsPerRow: CGFloat = 2.0
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: numberOfItems should be Int

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 26]
```diff
@@ -8,115 +8,149 @@
...
 
-class HomeViewController: ViewController {
+final class HomeViewController: ViewController {
 
-    @IBOutlet weak var tableView: UITableView!
-    @IBOutlet weak var collectionView: UICollectionView!
+    @IBOutlet weak var headerView: UIView!
+    @IBOutlet weak var listCollectionView: UICollectionView!
+    @IBOutlet weak var scrollView: UIScrollView!
+    fileprivate var images: [UIImage] = [#imageLiteral(resourceName: "miquang"), #imageLiteral(resourceName: "monan1"), #imageLiteral(resourceName: "monan2"), #imageLiteral(resourceName: "monan3"), #imageLiteral(resourceName: "monan4"), #imageLiteral(resourceName: "monan5")]
```
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: images should be belong to ViewModel

File: MyApp/View/Controllers/Home/HomeViewController.swift (hunk 24 above)
Comment (at-haiphung) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: should get data from VM

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 27]
```diff
@@ -22,6 +22,7 @@ final class HomeViewController: ViewController {
         configScroolView()
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Naming

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 28]
```diff
@@ -22,6 +22,7 @@ final class HomeViewController: ViewController {
...
         confignavigationBar()
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Naming

**Comment by at-hieutran** [comment]
LGTM

**Pull request merged**